Source/Applications/SystemCenter/wwwroot/Scripts/TSX/SystemCenter/ExternalDB/ExternalDBTableFields.tsx

```diff
@@ -0,0 +1,290 @@
+                <button className="btn btn-danger btn-block" onClick={() => { DisassociateField(record); setShowRemove(false); }}>Disassociate Field From This Table</button>
+                <button className="btn btn-danger btn-block" onClick={() => { Delete(); setShowRemove(false); }}>Delete Field Permanently</button>
+            </Modal>
+
+            <Modal Title={record.ID == 0 ? 'Add New Field' : 'Edit ' + (record?.FieldName ?? 'Field')} Show={showNew} ShowCancel={false} ConfirmText={record.ID == 0 ? 'Add' : 'Save'}
+                ConfirmShowToolTip={errors.length > 0 || warnings.length > 0}
+                ConfirmToolTipContent={
+                    <>
+                        { warnings.map((w, i) => <p key={i}>{Warning} {w}</p>) }
+                        { errors.map((e, i) => <p key={i}>{CrossMark} {e}</p>) }
+                    </>
+                }
+                DisableConfirm={errors.length > 0}
+                ShowX={true} CallBack={(conf) => {
+                    setShowNew(false);
+                    if (conf && record.ID > 0)
+                        dispatch(AdditionalFieldsSlice.DBAction({ verb: 'PATCH', record }));
+                    else if (conf && record.ID == 0)
+                        dispatch(AdditionalFieldsSlice.DBAction({ verb: 'POST', record }));
+                }}
+            >
+                <ExternalDBTableFieldForm Record={record} Setter={setRecord} SetErrors={setErrors} />
+            </Modal>
+
+            {/* TODO: for some reason, you have to click on a column key to sort the table before any of the addl fields are populated that weren't part of the original fields */}
+            <SelectPopup<SystemCenter.Types.AdditionalField>
+                Slice={AdditionalFieldsSlice}
+                Title={"Add Fields to " + (props.TableName ?? 'External DB Table')}
+                Selection={origFields}
+                OnClose={(selected, conf) => {
+                    setShowExisting(false);
+
+                    if (conf) {
+                        selected.filter((s) => origFields.findIndex((o) => o.ID === s.ID) < 0).forEach((f) => AssociateField(f));
+                        origFields.filter((o) => selected.findIndex((s) => s.ID === o.ID) < 0).forEach((f) => DisassociateField(f));
+                    }
+                }}
+                Show={showExisting}
+                Type={'multiple'}
+                Columns={[
+                    { key: 'FieldName', field: 'FieldName', label: 'Name', headerStyle: { width: 'auto' }, rowStyle: { width: 'auto' } },
+                    { key: 'ParentTable', field: 'ParentTable', label: 'Parent Type', headerStyle: { width: 'auto' }, rowStyle: { width: 'auto' } },
+                    { key: 'Type', field: 'Type', label: 'Type', headerStyle: { width: 'auto' }, rowStyle: { width: 'auto' } },
```

Comment:
Should we make this label `Field Type` to better differentiate from `Parent Type`?

Reply:
Or `Value Type` or something else. I'm not picky, just seems like something that could be confusing to some.

Reply:
Yeah `Field Type` sounds good